magento/magento2
# Remove Magento_ReleaseNotification module and cleanup all traces of it

### Description (*)
The release notification module has been broken since Magento 2.4.0 as explained over here: https://github.com/magento/magento2/issues/29363

Also after some more investigation, it still tries to fetch content from a remote endpoint on the magento.com domain that no longer seems to exist at the time of writing, for example: https://magento.com/release_notifications/2.2.4/community/en_us.json (I picked 2.2.4 because I know for sure that that release contained a release notification popup, as demonstrated in the second screenshot in https://github.com/magento/magento2/issues/29363#issuecomment-754913283)

So to me it sounds like this module won't be used anymore now and in the future.

Since this module was referenced in the `Magento_AdminAnalytics` module, I also removed all traces over there and had to rework a little bit of the css there to make the line height in the popup match with how it was before (since it referenced some css from the release notification module).

### Related Pull Requests

https://github.com/magento-gl/magento2-infrastructure/pull/16

### Fixed Issues (if relevant)
1. Fixes magento/magento2#29363

### Manual testing scenarios (*)
1. Since the removed module got referenced in the `Magento_AdminAnalytics` module, make sure that module still works properly
2. Easiest way to test is to temporarily modify the method `Magento\AdminAnalytics\Model\Condition\CanViewNotification::isVisible` and hardcode a `return true;` in its first line
3. After doing so, visit the dashboard in the backoffice and you should see a popup showing up, verify that it looks exactly like it looked before the removal of the ReleaseNotification module

### Questions or comments
Is this PR considered backwards compatible?
Or should we instead of completely removing the module, just hardcode it to never show up again in the backoffice, so we don't get that weird half-rendered popup anymore you see when visiting the backoffice for the first time on every single new release of Magento like shown in the first screenshot of https://github.com/magento/magento2/issues/29363#issuecomment-754913283 ?

### Contribution checklist (*)
